include/common/ttypes.h

```diff
@@ -141,6 +141,43 @@ typedef struct {
     }                                          \
   } while (0)
 
+#define NUM_TO_STRING(_inputType, _input, _outputBytes, _output)     \
```

Comment:
why not inline function? It's hard to maintain such a big "macro"

Reply:
Agreed big macro is hard to maintain. This macro is simply ported from 2.6 so pretty much following the same convention if look at other macros defined above this code block: e.g. SET_TYPED_DATA/GET_TYPED_DATA etc. 
Not sure if inlining this as function would be cost efficient as this code block is pretty large and contains many conditional branches(assume there's penalty for cpu to do branch prediction).

Reply:
ok, if it's ported code, let's keep it.